src/Config/Mapper.php

```diff
@@ -4,93 +4,91 @@
 
 namespace PharIo\ComposerDistributor\Config;
 
+use DOMDocument;
 use PharIo\ComposerDistributor\File;
 use PharIo\ComposerDistributor\FileList;
 use PharIo\ComposerDistributor\Url;
+use RuntimeException;
 
 class Mapper
 {
-    /**
-     * Expected format:
-     *
-     * 'keyDirectory' and 'signature' are optional
-     *
-     * [
-     *   'packageName'  => 'phar-io/phive',
-     *   'keyDirectory' => 'keys',
-     *   'phars'        => [
-     *     [
-     *       'name'      => 'phive',
-     *       'file'      => 'https://github.com/phar-io/phive/releases/%version%/phive.phar',
-     *       'signature' => 'https://github.com/phar-io/phive/releases/%version%/phive.phar.asc'
-     *     ]
-     *   ]
-     * ]
-     *
-     */
-    public function createConfig(array $configData): Config
+    /** @var \DOMDocument */
+    private $document;
+
+    public function createConfig(DOMDocument $docDocument): Config
     {
-        $this->validateBaseConfig($configData);
-        $this->validatePharsConfig($configData);
+        $this->document = $docDocument;
+        $this->validateConfigurationAgainstSchema();
 
         return new Config(
-            $configData['packageName'],
-            $this->createPhars($configData['phars']),
-            $this->createKeyDir($configData)
+            $this->packageName(),
+            $this->createPhars(),
+            $this->createKeyDir()
         );
     }
 
-    private function createKeyDir(array $configData): ?\SplFileInfo
+    private function validateConfigurationAgainstSchema()
     {
-        return isset($configData['keyDirectory'])
-            ? new \SplFileInfo($configData['keyDirectory'])
-            : null;
+        $original    = \libxml_use_internal_errors(true);
+        $xsdFilename = __DIR__ . '/../../composer-distributor.xsd';
+        $errors      = [];
+
+        $this->document->schemaValidate($xsdFilename);
+        foreach (\libxml_get_errors() as $error) {
+            if (!isset($errors[$error->line])) {
+                $errors[$error->line] = [];
+            }
+            $errors[$error->line][] = \trim($error->message);
+        }
+        \libxml_clear_errors();
+        \libxml_use_internal_errors($original);
+
+        if (count($errors) > 0) {
```

Comment:
Given that `DOMDocument::schemaValidate()` returns a `bool` to signal success or failure, we should be able to exit early and only continue with error handling in case of `false`.

Or not?

Reply:
Done